packages/playwright-test/src/worker/testInfo.ts

```diff
@@ -157,6 +161,13 @@ export class TestInfoImpl implements TestInfo {
       const relativeTestFilePath = path.relative(this.project.testDir, test._requireFile);
       return path.join(this.project.snapshotDir, relativeTestFilePath + '-snapshots');
     })();
+
+    this._attachmentsPush = this.attachments.push.bind(this.attachments);
+    this.attachments.push = (...attachments: TestInfo['attachments']) => {
```

Comment:
do we want to handle removal of an attachment from the list?

Reply:
I don't think so!